docker-compose.yaml

```diff
@@ -51,14 +51,41 @@ services:
     cap_add:
       - SYS_ADMIN
       - CAP_SYS_NICE
-    ports:
-      - 5500
   ceph-nvmeof:
     extends:
       service: ceph-nvmeof-base
     depends_on:
       ceph-cluster:
         condition: service_healthy
+    ports:
+      # Gateway RPC see addr/port in ceph-nvmeof.conf
+      - 0.0.0.0:5500:5500
+
+  ceph-nvmeof-pytest-base:
+    # Run tests code in current dir
+    extends:
+      service: ceph-nvmeof-base
+    volumes:
+      - ./tests:/src/tests
+
+  ceph-nvmeof-pytest-cli:
+    extends:
+      service: ceph-nvmeof-pytest-base
+    entrypoint: |
+      python3 -m pytest tests/test_cli.py
+
+  ceph-nvmeof-pytest-multi-gateway:
+    extends:
+      service: ceph-nvmeof-pytest-base
+    entrypoint: |
+      python3 -m pytest tests/test_multi_gateway.py
+
+  ceph-nvmeof-pytest-state:
+    extends:
+      service: ceph-nvmeof-pytest-base
+    entrypoint: |
+      python3 -m pytest tests/test_state.py
```

Comment:
Rather than per-PR containers, I'd have a lightweight Python-only container, or even better no container at all, and just use tox (it's still the default way of running isolated tests in Python). WDYT?